### PR TITLE
handle `--json-output` properly

### LINKED ISSUE #133855: ./x.py test doesn't respect --json-output for rustc error messages
`./x.py check` doesn't check tests for the standard library, so I have to use `./x.py test` instead when editing tests. Because `--json-output` is not respected for `./x.py test` by bootstrap, I can't get inline error messages in my ide.

### PR DESCRIPTION
**Pull request by onur-ozkan**
Because `rustfmt` doesn't support JSON output, `x test --json-output` doesn't respect the `--json-output` flag during formatting step. This change makes that `x test` skips the formatting step if `--json-output` is specified. In addition, resolves #133855 with the 2nd commit.

### CODE REVIEW

**Review by jieyouxu — APPROVED** [review]
Thanks! I tried this locally and this seems to work properly. You can r=me after the `eprintln` nit.

File: src/bootstrap/src/core/build_steps/test.rs [hunk 1]
```diff
@@ -1070,23 +1070,33 @@ impl Step for Tidy {
...
+                crate::core::build_steps::format::format(
+                    builder,
+                    !builder.config.cmd.bless(),
+                    all,
+                    &[],
+                );
+            } else {
+                println!(
+                    "WARNING: `--json-output` is not supported on rustfmt, formatting will be skipped"
                 );
```
[the remark this answers is not in the document]
Comment (onur-ozkan) [reply src/bootstrap/src/core/build_steps/test.rs:1098]: Sure

**Pull request merged**